src/net.h

```diff
@@ -759,8 +759,8 @@ class CNode
     // socket
     std::atomic<ServiceFlags> nServices{NODE_NONE};
     SOCKET hSocket GUARDED_BY(cs_hSocket);
-    size_t nSendSize{0}; // total size of all vSendMsg entries
-    size_t nSendOffset{0}; // offset inside the first vSendMsg already sent
+    size_t nSendSize GUARDED_BY(cs_vSend){0}; // total size of all vSendMsg entries
+    size_t nSendOffset GUARDED_BY(cs_vSend){0}; // offset inside the first vSendMsg already sent
     uint64_t nSendBytes GUARDED_BY(cs_vSend){0};
     std::deque<std::vector<unsigned char>> vSendMsg GUARDED_BY(cs_vSend);
     RecursiveMutex cs_vSend;
```

Comment:
I think this could use a comment. Something like:

```
//* Guards the send buffer and statistics about data sent on this connection */
```

Reply:
The `GUARDED_BY` annotation seems quite self-documented; especially, when the mutex and all of the data members protected by it are nicely grouped :)